src/box/box.cc

```diff
@@ -1506,6 +1509,22 @@ box_check_node_name(char *out, const char *cfg_name, bool set_diag)
 		}
 		return -1;
 	}
+
+	/**
+	 * If schema version is less then 3.0.0, then we consider setting name
+	 * as NoOp in order to allow painless configuration of Tarantool 3.0
+	 * on the lower schema . if dd_version_id == 0, then recovery/bootstrap
+	 * is not done, we can say nothing, allow retrieving a name.
+	 */
+	if (dd_version_id != 0 && dd_version_id < MIN_NAMES_VERSION_ID) {
+		say_warn("Setting %s was skipped. Please, consider upgrading "
+			 "schema. If you're using yaml configuration names "
+			 "will be automatically set after schema upgrade",
+			 cfg_name);
+		*out = 0;
+		return 0;
+	}
+
```

Comment:
We should probably also allow skip MISMATCH errors when in snaps there's nil value and force_recovery is set. This will allow to use yaml config for restoring instance, which failed to set names in box.watch trigger.

This doesn't contradict with NoOp rule, if names are set on 2.11, they don't work correctly, even though setting them is allowed without this commit. It's impossible to set them on 3.0, as they're supposed to be already set (saved in `cfg_instance_name` and so on).

Reply:
Instead of NoOp we may absolutely forbid specifying names, when schema version is < 3.0.0. The problem, is that we'll recover snaps, which can take the significant amount of time, and only after that we'll say user, that it's forbidden

Reply:
The option I like more - doesn't change the way, names works now at all. Do everything in `box/lua/config`